python/cugraph/cugraph/dask/cores/core_number.py

```diff
@@ -126,7 +127,16 @@ def core_number(input_graph,
 
     wait(cudf_result)
 
-    ddf = dask_cudf.from_delayed(cudf_result)
+    # FIXME: Dask doesn't always release it fast enough.
+    # For instance if the algo is run several times with
+    # the same PLC graph, the current iteration might try to cache
+    # the past iteration's futures and this can cause a hang if some
+    # of those futures get released midway
+    del result
+
+    ddf = dask_cudf.from_delayed(cudf_result).persist()
+    wait(ddf)
+
```

Comment:
I suggest we also remove `cudf_result` here to save on memory . We have a renumbered call after this so `result` does not go out of scope immediately.